### PR TITLE
Edit columns component and functionality

### PR DESCRIPTION
**Pull request by sauravmaiti22**
- Added a new component called EditColumns which is a headless UI popover primitive.
- This component accepts a list of columns that have an id and label. The id is useful for instances in future if we decide to rename a column, the id wouldn't change and the persisted selection will be valid.
- If the user tries to deselect all columns, then the last selected column checkbox will be disabled.
- The column selector is placed at the bottom of table as per new design which will have paging next to it - https://www.figma.com/file/Sc96Upmn3MpOmvM9jgvaUG/MMB-Portal-%2F-HiFi?type=design&node-id=11056%3A53890&mode=design&t=f3Wfhuu5ek6yxVKd-1. Will be implemented in a separate task.
- Have to add new columns to the table that will be deselected by default. Also have to implement horizontal scroll. Will be done in separate task. 
- Had to refactor the pr table to combine tags and context menu in a single `<td>` to let the last column take up all the remaining  space.
- Added conditional rendering of columns based on selection.
- Added a store to save selection of columns in local storage.

### CODE REVIEW

**Comment by sipsorcery** [comment]
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/178.

File: packages/components/src/components/ui/atoms/DropDown/DropDown.tsx [hunk 1]
```diff
@@ -91,8 +93,8 @@ DropdownMenuItem.displayName = DropdownMenuPrimitive.Item.displayName
 
 export {
   DropdownMenu,
+  DropdownMenuCheckboxItem,
   DropdownMenuContent,
-  //   DropdownMenuCheckboxItem,
```
Comment (donalnofrixion) [inline packages/components/src/components/ui/atoms/DropDown/DropDown.tsx:95]: Can these be removed?

**Review by donalnofrixion — APPROVED** [review]

**Comment by donalnofrixion** [comment]
Very nice! 🔥

File: packages/components/src/components/ui/atoms/DropDown/DropDown.tsx (hunk 1 above)
Comment (sauravmaiti22) [reply packages/components/src/components/ui/atoms/DropDown/DropDown.tsx:95]: Oh yes. Let me remove it.

**Review by laucait — APPROVED** [review]

**Review by pabloi-nf — CHANGES_REQUESTED** [review]
Looking good!

The only thing to adjust is the margins. I can see it has too much at the bottom.
image

I was making a guide for general pages and components in Figma that we can use as reference for this.
https://www.figma.com/file/Sc96Upmn3MpOmvM9jgvaUG/MMB-Portal-%2F-HiFi?type=design&node-id=11589%3A60410&mode=dev

Another thing is that it needs a distance between the icon and the opened panel. We should give 1rem distance to the edge of the icon.
image

**Review by arifmatin — APPROVED** [review]

**Comment by sipsorcery** [comment]
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/178.

**Review by pabloi-nf — APPROVED** [review]

**Pull request merged**